chorus_lib/src/core.rs

```diff
@@ -338,13 +333,16 @@ where
     where
         L: Subset<LS, IndexSet>,
     {
-        struct EppOp<'a, L: HList, L1: ChoreographyLocation, B: Transport> {
+        struct EppOp<'a, L: HList, L1: ChoreographyLocation, LS: HList, B: Transport<LS>> {
```

Comment:
Question: What's the difference between `L` and `LS`?

Reply:
Hmm; It seems to me that `L` here comes from the Choreography: Look at Line 329. The L first is inferred from the Choreography, then at line 439 we the same L as a concrete value for the L at line 336.

But LS is the location set for the transport (and the projector). So we need it as a different generic and can't use L. L is a subset of LS (line 334).

There is an issue where `LS` is first introduces on line 287, but has to be introduced again on line 343; and it seems it's Rust thing where you can't use a generic of an outer function inside an inner function!